### ISSUE TITLE
Godot 4 android custom build throws error

### ISSUE THREAD

**Issue by himaghnam** [godotengine/godot#56404]
### Godot version

4.0 dev 

### System information

win 10

### Issue description

I have JDK 11 installed with the proper environment and works fine while compiling plugins.
3.4 worked fine with java 11.

```
Task :compileDebugJavaWithJavac FAILED
D:\New folder (2)\demo\android\build\src\com\godot\game\GodotApp.java:33: error: package org.godotengine.godot does not exist
import org.godotengine.godot.FullScreenGodotApp;
                            ^
D:\New folder (2)\demo\android\build\src\com\godot\game\GodotApp.java:41: error: cannot find symbol
public class GodotApp extends FullScreenGodotApp {
                              ^
  symbol: class FullScreenGodotApp
D:\New folder (2)\demo\android\build\src\com\godot\game\GodotApp.java:42: error: method does not override or implement a method from a supertype
	@Override
	^
D:\New folder (2)\demo\android\build\src\com\godot\game\GodotApp.java:44: error: cannot find symbol
		setTheme(R.style.GodotAppMainTheme);
		^
  symbol:   method setTheme(int)
  location: class GodotApp
D:\New folder (2)\demo\android\build\src\com\godot\game\GodotApp.java:45: error: cannot find symbol
		super.onCreate(savedInstanceState);
		^
  symbol:   variable super
  location: class GodotApp
5 errors

FAILURE: Build failed with an exception.
```

### Steps to reproduce

Use custom build

**Comment by Calinou** [comment]
@himaghnam Please surround large blocks of text with triple backticks like this:

```
text here
```

This prevents them from taking too much space and also makes sure issue references aren't accidentally created. I edited your comment, but remember to do this next time :slightly_smiling_face:

**Comment by himaghnam** [comment]
Not having this issue in the latest pre-alpha, closing!

**Comment by Gromph** [comment]
I ran into this error with our custom build after upgrading from Godot 3.4 to 3.5.   After many hours, I tracked it down to we were building debug android binaries, ie:
scons platform platform=android target=debug bits=64 arch=arm64

Then building the template:
gradlew generateGodotTemplates

In godot 3.4 this would result in the android_source.zip template containing the file:
libs\debug\godot-lib.debug.aar

But in godot 3.5 the file is now:
libs\dev\godot-lib.dev.aar

Then when exporting it would result in the same errors posted above, because it was looking for godot-lib.debug.aar not godot-lib.dev.aar

Anyways the fix was to compile binaries with release_debug instead of debug:
scons platform platform=android target=debug bits=64 arch=arm64

The docs are correct, so probably don't need to re-open this issue.  Just wanted to put this here in case anyone else runs into this issue.
